### ISSUE TITLE
Получение настроек /GetSettings

### ISSUE THREAD

**Issue by nikitaryanov** [1C-Company/DirectBank#49]
Добрый день!

Возник вопрос по поводу алгоритма получения настроек:

1. В документации написано: 

> передача в банк расчетного счета Клиента, банк по этому номеру формирует файл настроек и отправляет обратно;

В то же время в самом запросе:

> Account: <Номер расчетного счета Клиента, для зарплатных проектов может отсутствовать>

Если расчетный счет клиента отсутствует, но при этом есть необходимость получения файла настроек, то корректно ли будет всегда использовать ИНН организации, так как customerId также может быть пустым (неизвестным) в теле запроса?

2. В файле настроек есть настройки для логина: Login/Certificate. Не совсем понятен смысл возврата этих параметров, если файл настроек можно получить уже будучи авторизованным одним из этих способов. Что тогда следует возвращать в этом случае? Тот же самый логин/сертификат, который был использован при авторизации, или что-то иное? 

3. В документах есть параметр, отвечающий за группы подписантов. Если групп больше 1, то верно ли утверждение, что это пользователи (подписанты) с разными логинами? 

4. Может ли быть такое, что по одному логину, в зависимости от файла настроек, полученных для конкретной организации, можно получать данные от разных организаций?

Спасибо!

**Comment by SevDmitry** [comment]
> Если расчетный счет клиента отсутствует, но при этом есть необходимость получения файла настроек, то корректно ли будет всегда использовать ИНН организации, так как customerId также может быть пустым (неизвестным) в теле запроса?

Да, лучше всего использовать ИНН. Некоторым банкам этого было недостаточно, поэтому добавили еще номер счета.

> В файле настроек есть настройки для логина: Login/Certificate. Не совсем понятен смысл возврата этих параметров, если файл настроек можно получить уже будучи авторизованным одним из этих способов. Что тогда следует возвращать в этом случае? Тот же самый логин/сертификат, который был использован при авторизации, или что-то иное?

Здесь нужно обратить внимание что это choice. Если из банка вернется элемент Login, то в дальнейшем при аутентификации программа запросит у пользователя ввести логин и пароль. Если вернется элемент Certificate, то при аутентификации будет запрошен выбор сертификата и пароль к нему. Таким образом пользователь может получить настройки обмена с аутентификацией по логину, а затем обмениваться с банком с аутентификацией по сертификату. Логин/сертификат аутентификации и в файле могут различаться, но это уже будет экзотикой.

> В документах есть параметр, отвечающий за группы подписантов. Если групп больше 1, то верно ли утверждение, что это пользователи (подписанты) с разными логинами?

Да, логины должны быть разные, но бывают и исключения. Т.е. логин выдан на организацию, но каждый пользователь имеет собственный сертификат подписи.

> Может ли быть такое, что по одному логину, в зависимости от файла настроек, полученных для конкретной организации, можно получать данные от разных организаций?

Да, по одному логину возможен обмен по разным организациям. Но 1С будет запрашивать пароль для каждой из них.

**Comment by nikitaryanov** [comment]
> Да, логины должны быть разные, но бывают и исключения. Т.е. логин выдан на организацию, но каждый пользователь имеет собственный сертификат подписи.

Получается, что к одной организации может относиться множество логинов пользователей или у организации один логин для всех и отличия между пользователями только в собственной подписи? Или вторая ситуация есть пример исключения?

**Comment by nikitaryanov** [comment]
@SevDmitry Спасибо за ответы. Возник еще один уточняющий вопрос: 

> Да, логины должны быть разные, но бывают и исключения. Т.е. логин выдан на организацию, но каждый пользователь имеет собственный сертификат подписи.

Получается, что к одной организации может относиться множество логинов пользователей или у организации один логин для всех и отличия между пользователями только в собственной подписи? Или вторая ситуация есть пример исключения?

**Comment by SevDmitry** [comment]
> Получается, что к одной организации может относиться множество логинов пользователей или у организации один логин для всех и отличия между пользователями только в собственной подписи? Или вторая ситуация есть пример исключения?

У одной организации может быть много логинов

**Comment by nikitaryanov** [comment]
Спасибо!

**Comment by nikitaryanov** [comment]
Добрый день!

При отправке выписки и декодирования ее на стороне 1с возникает ошибка:


Выполнение операции: Чтение данных электронного документа.
Возникла ошибка при чтении данных электронного документа.
Текст ошибки: Значение не является значением объектного типа (Sum)
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(8967)}:                РеквизитыПлатежа.Добавить(ОбменСБанкамиКлиентСервер.ЧислоИзСтроки(PayData.Sum), "Сумма");
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(8905)}:                                                           ПрочитатьОперациюВыписки(ЭД, Операция, НовыйЭД);
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(8520)}:                                                                                                           ПрочитатьВыпискуAsyncXDTO(ЭД, ДеревоРазбора, НовыйЭД);
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(6004)}:            ДеревоРазбора = ПрочитатьФайлПоСхеме(ВидЭД, ДанныеФайла);
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(10446)}:          СтруктураРазбораФайла = СформироватьДеревоРазбора(Перечисления.ВидыЭДОбменСБанками.ВыпискаБанка, ДвоичныеДанныеФайла);
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(10351)}:                          ПрочитатьВыпискуБанкаАсинхронныйОбмен(НастройкаОбмена, ПакетОбменСБанками, ДвоичныеДанныеФайла, СоздаватьОперацииВыписки, НовоеСообщениеОбмена);
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(4677)}:                                                           ПрочитатьДокументИзПакета(НастройкаОбмена, ПакетОбменСБанками, Document, МассивВнешнихИдентификаторов, СоздаватьОперацииВыписки, ДанныеВозврата);
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(4780)}:                            РаспаковатьПакетОбменСБанками(ПакетОбменСБанками, СоздаватьОперацииВыписки, ДанныеВозврата);
{ОбщийМодуль.ОбменСБанкамиСлужебный.Модуль(3986)}:                            СохранитьПолученныйПакет(Параметры.НастройкаОбмена, Результат.Путь, СоздаватьОперацииВыписки, ДанныеВозврата, ДанныеВозврата.ТребуетсяПовторнаяАутентификация);
{(1)}:ОбменСБанкамиСлужебный.ПолучитьНовыеДокументыИзБанка(Параметры[0],Параметры[1])
{ОбщийМодуль.ОбщегоНазначения.Модуль(4883)}:            Выполнить ИмяМетода + "(" + ПараметрыСтрока + ")";
{ОбщийМодуль.ДлительныеОперации.Модуль(982)}:                         ОбщегоНазначения.ВыполнитьМетодКонфигурации(ИмяПроцедуры, ПараметрыВызова);
{ОбщийМодуль.ДлительныеОперации.Модуль(972)}:                         ВызватьПроцедуру(ВсеПараметры.ИмяПроцедуры, ВсеПараметры.ПараметрыПроцедуры);

Файл C:\Users\Administrator\AppData\Local\Temp\v8_6162_28a.xml.

Не совсем понятна причина ее возникновения.
Значением компонента <Sum> является 0, что, судя по описанию типов, попадает под ограничение BigDecimal.

В чем может быть проблема?

Спасибо!


С Уважением,
Рьянов Никита


-----------------------------------
This message and any attachment are confidential and may be privileged or otherwise protected from disclosure. If you are not the intended recipient any use, distribution, copying or disclosure is strictly prohibited. If you have received this message in error, please notify the sender immediately either by telephone or by e-mail and delete this message and any attachment from your system. Correspondence via e-mail is for information purposes only. AO Raiffeisenbank neither makes nor accepts legally binding statements by e-mail unless otherwise agreed. 
-----------------------------------
